lib/createDeferredTask.js

```diff
@@ -0,0 +1,17 @@
+/**
+ * Create a deferred task that can be resolved when we call `resolve()`
+ * The returned promise will complete only when we call `resolve` or `reject`
+ * Useful when we want to wait for a tasks that is resolved from an external action
+ *
+ * @template T
+ * @returns {{ resolve: function(*), reject: function(Error), promise: Promise<T|void> }}
+ */
+export default function createDeferredTask() {
+    const deferred = {};
+    deferred.promise = new Promise((res, rej) => {
+        deferred.resolve = res;
+        deferred.reject = rej;
```

Comment:
Let's implement this or remove it?

Reply:
What do you mean the `reject` functionality?
It should be up to the caller - if they need to reject they should have the option exposed here

All the deferred tasks provide the reject/resolve/promise interface - this is the standard pattern

Reply:
>It should be up to the caller - if they need to reject they should have the option exposed here

I get what you are saying but I only see one caller. As a general rule, we either implement code or we remove it.